dbt/adapters/trino/connections.py

```diff
@@ -21,32 +22,201 @@
 logger = AdapterLogger("Trino")
 
 
+class TrinoCredentialsFactory:
+    @classmethod
+    def _create_trino_profile(cls, profile):
+        if "method" in profile:
+            method = profile["method"]
+            if method == "ldap":
+                return TrinoLdapCredentials
+            elif method == "certificate":
+                return TrinoCertificateCredentials
+            elif method == "kerberos":
+                return TrinoKerberosCredentials
+            elif method == "jwt":
+                return TrinoJwtCredentials
+            elif method == "oauth":
+                return TrinoOauthCredentials
+        return TrinoNoneCredentials
+
+    @classmethod
+    def translate_aliases(
+        cls, kwargs: Dict[str, Any], recurse: bool = False
+    ) -> Dict[str, Any]:
+        klazz = cls._create_trino_profile(kwargs)
+        return klazz.translate_aliases(kwargs, recurse)
+
+    @classmethod
+    def validate(cls, data: Any):
+        klazz = cls._create_trino_profile(data)
+        return klazz.validate(data)
+
+    @classmethod
+    def from_dict(cls, data: Any):
+        klazz = cls._create_trino_profile(data)
+        return klazz.from_dict(data)
+
+
+class HttpScheme(Enum):
+    HTTP = "http"
+    HTTPS = "https"
+
+
 @dataclass
-class TrinoCredentials(Credentials):
+class TrinoNoneCredentials(Credentials):
     host: str
     port: Port
     user: str
-    password: Optional[str] = None
-    jwt_token: Optional[str] = None
-    client_certificate: Optional[str] = None
-    client_private_key: Optional[str] = None
+    cert: Optional[str] = None
     method: Optional[str] = None
+    http_scheme: HttpScheme = HttpScheme.HTTP
```

Comment:
```python
    @property
    def http_scheme(self):
        return HttpScheme.HTTP

Reply:
You could theoretically configure a Trino server with https but without any authentication configured. This is also documented as such.